onnx/defs/math/defs.cc

```diff
@@ -3412,4 +3412,594 @@ ONNX_OPERATOR_SET_SCHEMA(
+                  "The output has the shape: [size].",
+                  "T2",
+                  OpSchema::Single,
+                  true,
+                  1,
+                  OpSchema::NonDifferentiable);
+    schema.TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+      auto size = get_scalar_value_from_tensor<int64_t>(ctx.getInputData(0));
+      if (size > 0) {
+          ONNX_NAMESPACE::TensorShapeProto result_shape;
+          result_shape.add_dim()->set_dim_value(size);
+          updateOutputShape(ctx, 0, result_shape);
+      }
+      auto output_datatype = getAttribute(ctx, "output_datatype", static_cast<int64_t>(TensorProto_DataType_FLOAT));
+      updateOutputElemType(ctx, 0, output_datatype);
+    });
+  };
+}
+
+ONNX_OPERATOR_SET_SCHEMA(
+    HannWindow,
+    16,
+    OpSchema()
+        .FillUsing(CosineSumWindowOpDocGenerator("Hann"))
+        .TypeConstraint(
+            "T1",
+            {"tensor(int64)"},
+        "Constrain the input size to int64_t.")
+        .TypeConstraint(
+            "T2",
+            OpSchema::all_numeric_types_with_bfloat(),
+            "Constrain output types to numeric tensors.")
+        .FunctionBody(R"ONNX(
+        {
+          A0 = Constant <value = float {0.5}>()
+          A1 = Constant <value = float {0.5}>()
+          A2 = Constant <value = float {0.0}>()
+          Zero = Constant <value = float {0.0}>()
+          One = Constant <value = float {1.0}>()
+          Two = Constant <value = float {2.0}>()
+          Tau = Constant <value = float {6.2831853}>()
+          AngularIncrement = Div (Tau, size)
+          Range = Range (Zero, size, One)
+          RangeAngular = Mul (Range, AngularIncrement)
+          TwoRangeAngular = Mul (RangeAngular, Two)
+          CosTwoRangeAngular = Cos (TwoRangeAngular)
+          A2_Component = Mul (A2, CosTwoRangeAngular)
+          CosRangeAngular = Cos (RangeAngular)
+          A1_Component = Mul (A1, CosRangeAngular)
+          output = Add (A1_Component, A2_Component)
+          output = Sub (A0, output)
+        }
+        )ONNX"
+        ));
+
+ONNX_OPERATOR_SET_SCHEMA(
+    HammingWindow,
+    16,
+    OpSchema()
+        .FillUsing(CosineSumWindowOpDocGenerator("Hamming"))
+        .TypeConstraint(
+            "T1",
+            {"tensor(int64)"},
+        "Constrain the input size to int64_t.")
+        .TypeConstraint(
+            "T2",
+            OpSchema::all_numeric_types_with_bfloat(),
+            "Constrain output types to numeric tensors.")
+        .FunctionBody(R"ONNX(
+        {
+          A0 = Constant <value = float {0.54347826087}>()
+          A1 = Constant <value = float {0.45652173913}>()
+          A2 = Constant <value = float {0.0}>()
+          Zero = Constant <value = float {0.0}>()
+          One = Constant <value = float {1.0}>()
+          Two = Constant <value = float {2.0}>()
+          Tau = Constant <value = float {6.2831853}>()
+          AngularIncrement = Div (Tau, size)
```

Comment:
size is expected to be a float here. I assume size is the only input defined as an integer.

Reply:
cast to float